### PR TITLE
w3crawl.py initial tests

### PR DESCRIPTION
**Pull request by npdoty**
partial completion of #342, adds tests for the step of actually parsing an archived email page from HTML into an mboxMessage (doesn't cover the step of crawling all the message urls from different dated archive pages)

also fixes some docstring issues and warnings within w3crawl.py

### CODE REVIEW

**Comment by npdoty** [comment]
I'll need to rebase to use `unittest` as is now done in master.

**Comment by npdoty** [comment]
Okay, these tests are updated to use `unittest`. @sbenthall, this is ready for you to take a look.

**Comment by npdoty** [comment]
Okay, these tests now merge in the changes from the py3 conversion, fix a couple issues from that conversion with string encoding, and add tests to catch it in case of future regression.

File: tests/bigbang_test.py [hunk 1]
```diff
@@ -224,3 +225,16 @@ def test_labeled_blockmodel(self):
...
             msg="Incorrected edges in labeled blockmodel")
+
+class TestW3crawl(unittest.TestCase):
+    def test_w3c_archive_message_parsing(self):
+        test_html_path = os.path.join(CONFIG.test_data_path, 'w3crawl-test-message.html')
+        f = open(test_html_path, 'r')
+        html = f.read()
+        message = w3crawl.W3cMailingListArchivesParser().parsestr(html)
+
+        assert len(message.get_from()) > 0, "message doesn't have a From address"
```
Comment (sbenthall) [inline tests/bigbang_test.py:236]: see unittest docs for its supported assertion syntex. I.e.

self.assertGreater(X, Y)

https://docs.python.org/3/library/unittest.html

**Comment by sbenthall** [comment]
Looks good. tests pass.
using the unittest assertion syntax in the test would be nice.
I'll merge in a couple days if that change is not made, as it's not a blocker.

**Comment by npdoty** [comment]
surpassed by #381

**Pull request closed without merging**